airflow/utils/log/bluecore_bq_gcs_task_handler.py

```diff
@@ -0,0 +1,200 @@
+        elif ti.operator == "KubernetesJobOperator":
+            # If remote file is not available and the job is a Kubernetes Job,
+            # use the task instance attributes to query BigQuery logs and
+            # make the logs available in real time from the "Log" page of the UI.
+
+            # TODO:  job_name is always none on the first try, should fix this so an exception doesn't show each time
+            job_name = ti.xcom_pull(task_ids=ti.task_id, key='kubernetes_job_name')
+            if job_name is None:
+                raise Exception("There is no XCOM kubernetes_job_name. Something is exploding.")
+
+            pod_output = BQGCSTaskHandler.get_pods(job_name)
+
+            billing_project_id = configuration.get('core', 'kubernetes_bigquery_billing_project')
+            client = bigquery.Client(billing_project_id)
+
+            # get info for the BQ table name
+            data_project_name = configuration.get('core', 'kubernetes_bigquery_data_project')
+            # get BQ table's date (today).. based on insert time & not execution time.
+            bq_table_date = datetime.utcnow().date().strftime("%Y%m%d")
+            log_lines = []
+            tables = []
+
+            for pod in pod_output['items']:
+                pod_id = pod['metadata']['name']
+                for container in pod['spec']['containers']:
+                    container_name = string.replace(container['name'], '-', '_')
+                    # TODO: replace gke_logs with a variable
+                    bq_table_name = "{}.gke_logs.{}_{}".format(str(data_project_name), str(container_name),
+                                                               str(bq_table_date))
+                    tables.append((str(bq_table_name), str(pod_id)))
+
+            query = ("\n UNION ALL \n ".join(
+                [BQGCSTaskHandler.generate_query(bq_table_name=bq_table_name, pod_id=pod_id) for
+                 bq_table_name, pod_id in tables]) + "\n ORDER BY pod_id, logName, timestamp")
+
+            query_job = client.query(query)
+            result = query_job.result()
+
+            for row in result:
+                log_lines.append(str("{}  {}".format(row.logName, row.textPayload)))
+
+            return "\n".join(log_lines)
+
+        # else statement from gcs_task_handler.py
+        else:
+            log = super(BQGCSTaskHandler, self)._read(ti, try_number)
+            return log
+
+
+
+    @staticmethod
+    def get_pods(job_name):
+        """
+        Return pods for a given job_name
+
+        :param job_name: a unique job name
+        :type: string
+        :return: result form kubectl command that contains job pod and container information
+        :type: dict
+
+        """
+        return json.loads(BQGCSTaskHandler.retryable_check_output(args=[
+            'kubectl', 'get', 'pods', '-o', 'json', '-l', 'job-name==%s' % job_name]))
+
+
+    @staticmethod
+    def retryable_check_output(args, retry_count=3):
```

Comment:
I'd rather pull this out from kubernetes_operator.py and into a util module that is referenced here and there.